extensions.cabal

```diff
@@ -80,14 +82,23 @@ library
                          Extensions.Package
                          Extensions.Types
 
-  build-depends:       bytestring >= 0.10 && < 0.12
-                     , Cabal >= 3.0 && < 3.9
+  build-depends:       bytestring >= 0.10 && < 0.13
+                       -- We need to pin a single major version of
+                       -- Cabal here because the main reason we use
+                       -- Cabal is for its list of extensions.  Later
+                       -- versions have strictly more extensions, and
+                       -- we'll have missing patterns if we try to
+                       -- support more than one major version.  If
+                       -- this causes problems in practice let's
+                       -- revisit this decision and come up with
+                       -- another approach.
+                     , Cabal ^>= 3.10
```

Comment:
Spinned off a discussion in #111.

This hit me when trying to package `extensions` @ the latest for ArchLinux which is still on GHC 9.4.8 (unfortunately). I had to pin the package to `0.1.0.0`.